batchflow/tests/config_basic_test.py

```diff
@@ -0,0 +1,349 @@
+    assert config.flatten() == exp_flat
+
+    #Mixed-structured list initialization
+    init_list = [('a', None), ('b/c', 2), ('b', {'d' : 3})]
+    exp_flat = {'a': None, 'b/c': 2, 'b/d': 3}
+    config = Config(init_list)
+    assert config.flatten() == exp_flat
+
+    #Config-structured list initialization
+    init_list = [('a', Config({'b' : 2}))]
+    exp_flat = {'a/b': 2}
+    config = Config(init_list)
+    assert config.flatten() == exp_flat
+
+def test_list_init_bad():
+    """
+    Tests Config.__init__() using BAD input of list type.
+    """
+
+    #Int-keyed list initialization
+    init_list = [(0, 1)]
+    with pytest.raises(Exception):
+        Config(init_list)
+
+    #Bool-keyed list initialization
+    init_list = [(False, True)]
+    with pytest.raises(Exception):
+        Config(init_list)
+
+    #Bad-shaped list initialization
+    init_list = [('a', 0, 1)]
+    with pytest.raises(Exception):
+        Config(init_list)
+
+def test_config_init():
+    """
+    Tests Config.__init__() using input of Config type.
+    For inner structure check Config.flatten() is used.
+    """
+
+    #Basically, there nothing to test here,
+    #but since Config can be initialized with its own instance...
+    init_config = Config({'a': 0})
+    exp_flat = {'a' : 0}
+    config = Config(init_config)
+    assert config.flatten() == exp_flat
+
+def test_pop():
+    """
+    Tests Config.pop(), comparing the return value with expected one.
+    For inner structure check Config.flatten() is used.
+    """
+
+    #Pop scalar value by slashed-structured key
+    config = Config({'a' : {'b' : 1}})
+    pop_key = 'a/b'
+    exp_ret = 1
+    exp_flat = {'a' : {}}
+    assert config.pop(pop_key) == exp_ret
+    assert config.flatten() == exp_flat
+
+    #Pop dict value by simple key
+    config = Config({'a' : {'b' : 1}})
+    pop_key = 'a'
+    exp_ret = {'b' : 1}
+    exp_flat = {}
+    assert config.pop(pop_key) == exp_ret
+    assert config.flatten() == exp_flat
+
+def test_get():
+    """
+    Tests Config.get(), comparing the return value with expected one.
+    For inner structure check Config.flatten() is used.
+    """
+    #Get scalar value by slashed-structured key
+    config = Config({'a' : {'b' : 1}})
+    get_key = 'a/b'
+    exp_ret = 1
+    exp_flat = {'a/b' : 1}
+    assert config.get(get_key) == exp_ret
+    assert config.flatten() == exp_flat
+
+    #Get dict value by simple key
+    config = Config({'a' : {'b' : 1}})
+    get_key = 'a'
+    exp_ret = {'b' : 1}
+    exp_flat = {'a/b' : 1}
+    assert config.get(get_key) == exp_ret
+    assert config.flatten() == exp_flat
+
+def test_put():
+    """
+    Tests Config.put(), placing value by key in Config instance.
+    For inner structure check Config.flatten() is used.
+    """
+
+    #Put scalar value by simple key
+    config = Config({})
+    put_key = 'a'
+    put_val = 1
+    exp_flat = {'a' : 1}
+    config.put(put_key, put_val)
+    assert config.flatten() == exp_flat
```

Comment:
`put` might overwrite the whole config with a new key

Reply:
Corrected by applying put() to more complex Config instances.